src/core/ReactUpdateQueue.js

```diff
@@ -40,20 +41,27 @@ function getInternalInstanceReadyForUpdate(publicInstance, callerName) {
   );
 
   var internalInstance = ReactInstanceMap.get(publicInstance);
-  invariant(
-    internalInstance,
-    '%s(...): Can only update a mounted or mounting component. ' +
-    'This usually means you called %s() on an unmounted ' +
-    'component.',
-    callerName,
-    callerName
-  );
-  invariant(
-    internalInstance !== ReactLifeCycle.currentlyUnmountingInstance,
-    '%s(...): Cannot call %s() on an unmounting component.',
-    callerName,
-    callerName
-  );
+  if (!internalInstance) {
+    if (__DEV__) {
+      // Only warn when we have a callerName. Otherwise we should be silent.
+      // We're probably calling from enqueueCallback. We don't want to warn
+      // there because we already warned for the corresponding lifecycle method.
+      warning(
+        !callerName,
```

Comment:
I know. I wasn't wild about it, but I didn't want to duplicate the same checks in `enqueueCallback`. I thought about adding another argument to `getInternalInstanceReadyForUpdate` but got "clever".